src/FixedDataTableCell.js

```diff
@@ -218,22 +220,55 @@ class FixedDataTableCell extends React.Component {
     }
 
     var content;
-    if (
-      this.props.isHeader &&
-      (this.props.onColumnResizeEnd ||
-        this.props.onColumnReorderEnd)
-    ) {
+    if (this.props.isHeader && (this.props.onColumnResizeEnd || this.props.onColumnReorderEnd)) {
       // NOTE: Use plugins manually for backward compatibility. Will be removed in future release.
-      content = (
-        <ResizeReorderCell
-          {...cellProps}
-          onColumnReorderStart={this.props.onColumnReorderStart}
-          onColumnResizeEnd={this.props.onColumnResizeEnd}
-          onColumnReorderEnd={this.props.onColumnReorderEnd}
-        >
-          {props.cell}
-        </ResizeReorderCell>
-      );
+      if (this.props.onColumnResizeEnd && this.props.onColumnReorderEnd) {
+        content = (
+          <ReorderCell
+            {...cellProps}
+            onColumnReorderStart={(/*string*/columnKey) => {
+              this.props.toggleCellsRecycling(false, columnKey);
+            }}
+            onColumnReorderEnd={(/*object*/val) => {
+              this.props.toggleCellsRecycling(true);
+              this.props.onColumnReorderEnd(val);
+            }}
+          >
+            <ResizeCell
+              onColumnResizeEnd={this.props.onColumnResizeEnd}
+            >
+              {props.cell}
+            </ResizeCell>
+          </ReorderCell>
+        );
+      } else if (this.props.onColumnReorderEnd) {
+        content = (
+          <ReorderCell
+            {...cellProps}
+            onColumnReorderStart={(/*string*/columnKey) => {
+              this.props.toggleCellsRecycling(false, columnKey);
+            }}
+            onColumnReorderEnd={(/*object*/val) => {
+              this.props.toggleCellsRecycling(true);
+              this.props.onColumnReorderEnd(val);
+            }}>
+            {props.cell}
+          </ReorderCell>
+        );
+      } else {
+        cellProps = {
```

Comment:
does ResizeCell need these extra props? If so, won't the above render also need it?

Reply:
Yes, minWidth and maxWidth are required for ResizeCell. Other cell's won't need it as it is not useful for them